### PR TITLE
Disallow modification on frozen config

### PR DESCRIPTION
**Pull request by qcam**
I think we shall disallow modifying configs when it has been frozen or finalized

### CODE REVIEW

**Comment by AMHOL** [comment]
Thanks @qcam, looks good, could you update `CHANGELOG.md` please

**Comment by qcam** [comment]
Thanks @AMHOL, CHANGELOG updated and commits squashed.

**Comment by AMHOL** [comment]
Thanks @qcam 

Sorry I didn't think before but it might be better to define a custom error for this in https://github.com/dry-rb/dry-configurable/blob/master/lib/dry/configurable/error.rb, what do you think?

**Comment by qcam** [comment]
Custom error sounds cool to me! 👍 

Do you think we want a custom `freeze` method like `finalize!` in dry-container?

**Comment by AMHOL** [comment]
Yeah I think that would be good, could just call `freeze` and `@config.freeze`

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,5 +4,7 @@
 
 * Allow for boolean false as default setting value ([yuszuv](https://github.com/yuszuv))
 * Convert nested configs to nested hashes with `Config#to_h` ([saverio-kantox](https://github.com/saverio-kantox))
+* Introduce `Configurable.finalize!` which freezes config and its dependencies ([qcam](https://github.com/qcam))
```
Comment (AMHOL) [inline CHANGELOG.md]: Can you add this under a `## Added` title please

P.S. Sorry to be a pain and keep asking for changes

Comment (qcam) [inline CHANGELOG.md]: @AMHOL no worries, changelog updated as requested

**Pull request merged**